### PR TITLE
Adds support for symbol tag of a style within a label

### PR DESCRIPTION
**Pull request by mholthausen**
This MR

- enables to convert a `SYMBOL` defined within the `LABEL` as `STYLE`
@KaiVolland @jansule please review

### CODE REVIEW

File: src/mapfile2js/parse/resolveSymbolset.ts [hunk 1]
```diff
@@ -53,6 +53,27 @@ export function resolveSymbolsFromMapfile(mapfile: Mapfile): Mapfile {
...
               }
             });
+          }  else if (mclass.labels) {
+            // parse symbol data within a style tag of a label
+            mclass.labels.forEach((label) => {
+              const styles: MapfileStyle[] = label.styles as MapfileStyle[];
+              if (styles) {
+                styles.forEach((style: MapfileStyle) => {
+                  if (style.symbol) {
+                    symbols.forEach((symbol: any) => {
```
Comment (KaiVolland) [inline src/mapfile2js/parse/resolveSymbolset.ts:63]: You might use optional chaining here (`?.`)

**Review by KaiVolland — APPROVED** [review]
Just one minor none blocking note.

File: src/mapfile2js/parse/resolveSymbolset.ts [hunk 2]
```diff
@@ -59,18 +59,16 @@ export function resolveSymbolsFromMapfile(mapfile: Mapfile): Mapfile {
               const styles: MapfileStyle[] = label.styles as MapfileStyle[];
               if (styles) {
```
Comment (KaiVolland) [inline src/mapfile2js/parse/resolveSymbolset.ts:60]: ... same

File: src/mapfile2js/parse/resolveSymbolset.ts (hunk 1 above)
Comment (mholthausen) [reply src/mapfile2js/parse/resolveSymbolset.ts]: Changed into an optional chaining now.

**Comment by mholthausen** [comment]
Thanks for review!

File: src/MapfileStyleParser.ts [hunk 3]
```diff
@@ -824,6 +833,29 @@ export class MapfileStyleParser implements StyleParser {
...
+    mapfileLabel: MapfileLabel
+  ): IconSymbolizer[] | void {
+    const labelStyles: IconSymbolizer[] = [];
+    mapfileLabel?.styles?.forEach((style: MapfileStyle) => {
+      if (style.symbol) {
+        labelStyles.push(this.getIconSymbolizerFromMapfileStyle(style));
+      }
+    });
+
+    if (labelStyles.length) {
```
Comment (jansule) [inline src/MapfileStyleParser.ts:852]: I would prefer to just return an empty array and do the check in the calling method

File: src/MapfileStyleParser.ts [hunk 4]
```diff
@@ -810,6 +810,15 @@ export class MapfileStyleParser implements StyleParser {
       mapfileLayerLabelItem = mapfileClass.text ? mapfileClass.text : mapfileLayerLabelItem;
       mapfileClass.labels.forEach((mapfileLabel) => {
         mapfileLabel.text = mapfileLabel.text ? mapfileLabel.text : mapfileLayerLabelItem;
+
+        // Set Icons in front of the associated label
+        const labelIcons =  this.getIconsFromMapfileLabel(mapfileLabel);
+        if (labelIcons) {
```
Comment (jansule) [inline src/MapfileStyleParser.ts:816]: this can be removed with the change suggested below

File: src/MapfileStyleParser.ts [hunk 5]
```diff
@@ -421,7 +421,7 @@ export class MapfileStyleParser implements StyleParser {
   getIconSymbolizerFromMapfileStyle(mapfileStyle: MapfileStyle): IconSymbolizer {
     const iconSymbolizer: IconSymbolizer = {
       kind: 'Icon',
-      image: mapfileStyle.symbol.image,
+      image: mapfileStyle.symbol.image || mapfileStyle.symbol,
```
Comment (jansule) [inline src/MapfileStyleParser.ts:424]: please undo this change and adjust the caller accordingly

**Review by jansule — CHANGES_REQUESTED** [review]

File: src/MapfileStyleParser.ts (hunk 3 above)
Comment (mholthausen) [reply src/MapfileStyleParser.ts]: I have adjusted the return process.

File: src/MapfileStyleParser.ts (hunk 4 above)
Comment (mholthausen) [reply src/MapfileStyleParser.ts]: Removed the `if`

File: src/MapfileStyleParser.ts [hunk 6]
```diff
@@ -824,6 +831,24 @@ export class MapfileStyleParser implements StyleParser {
...
   }
 
+  /**
+   * Collect icon data if defined within the label tag
+   *
+   * @param {MapfileLabel} mapfileLabel Mapfile label data
+   * @return {IconSymbolizer[]} The IconSymbolizer
+   */
+  getIconsFromMapfileLabel(mapfileLabel: MapfileLabel): IconSymbolizer[] {
+    const labelStyles: IconSymbolizer[] = [];
```
Comment (jansule) [inline src/MapfileStyleParser.ts:841]: very minor and definitely non blocking: shouldn't this rather be `iconStyles`?

File: src/MapfileStyleParser.ts [hunk 7]
```diff
@@ -810,6 +810,13 @@ export class MapfileStyleParser implements StyleParser {
       mapfileLayerLabelItem = mapfileClass.text ? mapfileClass.text : mapfileLayerLabelItem;
       mapfileClass.labels.forEach((mapfileLabel) => {
         mapfileLabel.text = mapfileLabel.text ? mapfileLabel.text : mapfileLayerLabelItem;
+
+        // Set Icons in front of the associated label
+        const labelIcons = this.getIconsFromMapfileLabel(mapfileLabel);
+        labelIcons?.forEach(labelIcon => {
```
Comment (jansule) [inline src/MapfileStyleParser.ts:816]: Minor: `labelIcons?.forEach` -> `labelIcons.forEach`. `getIconsFromMapfileLabel` always returns an array.

**Review by jansule — APPROVED** [review]
Nice work @mholthausen. Just some nitpicks you might want to address

File: src/MapfileStyleParser.ts (hunk 6 above)
Comment (mholthausen) [reply src/MapfileStyleParser.ts]: Renamed the variable.

File: src/MapfileStyleParser.ts (hunk 7 above)
Comment (mholthausen) [reply src/MapfileStyleParser.ts]: Thx, removed the optional mark.
